src/core/public/styles/rendering/_base.scss

```diff
@@ -67,7 +67,7 @@
 }
 
 .kbnBody--chromeHidden {
-  --euiFixedHeadersOffset: 0;
+  --euiFixedHeadersOffset: 0px;
```

Comment:
The problem was: this calculation, which is used in Discover, was coming up wrong:
```
@media only screen and (min-width: 1200px) {
    .dscPage {
        // --kbnAppHeadersOffset is undefined
        // --euiFixedHeadersOffset is `0`
        height: calc(100vh - var(--kbnAppHeadersOffset, var(--euiFixedHeadersOffset, 0)));
    }
}
```

@cee-chen found the reason why in a spec: https://drafts.csswg.org/css-values-3/#calc-type-checking
> Note: Because [<number-token>](https://drafts.csswg.org/css-syntax-3/#typedef-number-token)s are always interpreted as [<number>](https://drafts.csswg.org/css-values-3/#number-value)s or [<integer>](https://drafts.csswg.org/css-values-3/#integer-value)s, "unitless 0" [<length>](https://drafts.csswg.org/css-values-3/#length-value)s aren’t supported in [calc()](https://drafts.csswg.org/css-values-3/#funcdef-calc). That is, [width: calc(0 + 5px);](https://drafts.csswg.org/css-sizing-3/#propdef-width) is invalid, even though both width: 0; and width: 5px; are valid.

Reply:
that's very strange, it seems that this change now causes a functional test error here:

https://github.com/elastic/kibana/blob/2b344d27979009300a0fdc9e8aa18aff28a9a66a/test/functional/services/visualizations/elastic_chart.ts#L43-L50

I've checked the html of the failure
![Bildschirmfoto 2024-09-26 um 09 51 49](https://github.com/user-attachments/assets/2cc2f7af-51ed-45bb-8573-51de28dfff91)
And so it failes because it seems `data-ech-render-complete=false` doesn't turn to `true`

dear @elastic/kibana-visualizations , do you have an idea what's happening here?

Reply:
Interesting. Does the test fail with the attribute still set to false, but the chart rendered? Or the panel is still in a loading phase?

Reply:
For me it looks like it was rendered?
![image](https://github.com/user-attachments/assets/9458cd37-c136-4ea2-a61a-eb7b3ee41ac3)

Reply:
Interesting, looks like the charts now starts a redraw due to the height change.
Is not the first time I see this, where the discover heatmap is rendered 2 times because of some jumps in the layout rendering of the table.

Reply:
hey @tsullivan to me it looks like something is happening between these lines

https://github.com/elastic/kibana/blob/f6f77b9a0bd63666c4934e107f46c1e5c73325a0/test/functional/apps/discover/group1/_discover_histogram.ts#L70-L86

Because from the looks I see that the chart correctly renders
```
[00:04:17]         │ debg renderings before brushing - actual: 1 expected: 1
```

but then before the `brushHistogram()` call something triggered a chart rerender. 
it could be that your parameter change actually triggered a different height size (causing the chat to try to rerender).

You could try to increase the timeout of the `brushHistogram` action, but I'm not sure this could solve the issue

Reply:
we could also try to remove the

Reply:
that's is a possible workaround. I believe there is still something that needs to be anyway solved as it doesn't seems to work correctly.
I've checked locally and I can see the following:
before this PR, when switching to full screen the chart height goes to zero, and back to its height when toggling the fullscreen off. With this PR the chart remain the same size but is hidden behind the position:fixed table. This, for some reason I don't understand cause a trigger in the ResizeObserver we are using to compute the chart size causing it to turn the rendered off, but doesn't trigger it again. I can investigate a bit more if you want

Reply:
ok, the way I see it, yes, it's related to the data grid going to full screen, and with the changes of this PR, as you said, the chart remains same size. For me it seems like we could try to remove the ` await this.testSubjects.click('dataGridFullScreenButton'); 
`, check for flakiness, and if there's no quick fix for the ResizeObserver not being triggered again  (which shold be investigated I think), open an issue to track this

Reply:
@kertal @tsullivan not sure if this is a bug in Charts or in ResizeObserver (that triggers a size change even if there is no size change in the chart), but I've created a PR to fix this in charts.
I will merge the fix in charts, and push an upgraded charts version in  Kibana today if possible

https://github.com/elastic/elastic-charts/pull/2534

Reply:
Awesome 🎉  thx!

Reply:
Sorry for the late work on that, we finally manage to fix the bug in charts, and we are now waiting for a CI to pass before merging the charts dependency update in Kibana https://github.com/elastic/kibana/pull/194794
Please do a bootstrap locally if you want to see and test the changes